File: scripts/pull-main.py
import subprocess
import os

os.chdir('/vercel/share/v0-project')

try:
    print('[v0] Fetching latest changes from remote...')
    subprocess.run(['git', 'fetch', 'origin'], check=True)
    
    print('[v0] Checking out main branch...')
    subprocess.run(['git', 'checkout', 'main'], check=True)
    
    print('[v0] Pulling latest changes from main...')
    subprocess.run(['git', 'pull', 'origin', 'main'], check=True)
    
    print('[v0] Checking current status...')
    result = subprocess.run(['git', 'status'], capture_output=True, text=True, check=True)
    print(result.stdout)
    
    print('[v0] Successfully updated from main branch!')
    
except subprocess.CalledProcessError as error:
    print(f'[v0] Error during git operations: {error}')
    exit(1)

File: scripts/pull-main.js
import { execSync } from 'child_process';

try {
  console.log('[v0] Starting git pull from main branch...');
  
  // Fetch all branches to ensure we have the latest
  execSync('git fetch origin', { cwd: '/vercel/share/v0-project', stdio: 'inherit' });
  console.log('[v0] Fetch completed');
  
  // Pull from main branch
  execSync('git pull origin main', { cwd: '/vercel/share/v0-project', stdio: 'inherit' });
  console.log('[v0] Pull from main completed successfully');
  
  // Show current status
  const status = execSync('git status', { cwd: '/vercel/share/v0-project', encoding: 'utf-8' });
  console.log('[v0] Current git status:\n', status);
  
} catch (error) {
  console.error('[v0] Error during git pull:', error.message);
  process.exit(1);
}
